File: app/src/main/java/com/robotnec/budget/core/mvp/presenter/CategoryOverviewPresenter.java
package com.robotnec.budget.core.mvp.presenter;

import com.robotnec.budget.core.di.ApplicationComponent;
import com.robotnec.budget.core.domain.Category;
import com.robotnec.budget.core.domain.operation.Transaction;
import com.robotnec.budget.core.mvp.view.CategoryOverviewView;
import com.robotnec.budget.core.persistence.dao.TransactionDao;
import com.robotnec.budget.core.service.aggregation.AggregationService;
import com.robotnec.budget.core.service.aggregation.impl.TransactionAggregation;

import java.util.List;

import javax.inject.Inject;

/**
 * @author zak <zak@swingpulse.com>
 */
public class CategoryOverviewPresenter extends Presenter<CategoryOverviewView> {

    @Inject
    TransactionDao transactionDao;

    @Inject
    AggregationService aggregationService;

    private final Category category;

    public CategoryOverviewPresenter(CategoryOverviewView view, Category category) {
        super(view);
        this.category = category;
    }

    @Override
    public void injectComponent(ApplicationComponent applicationComponent) {
        applicationComponent.inject(this);
    }

    @Override
    public void onViewResume() {
        List<Transaction> categoryTransactions = transactionDao.getTransactionsForCategory(category.getId());
        TransactionAggregation aggregation =
                aggregationService.aggregate(categoryTransactions, AggregationService.Resolution.DAY);
        view.displayCategoryTransactions(aggregation);
        view.displayCategoryTitle(category.getName());
    }
}

File: app/src/main/java/com/robotnec/budget/core/mvp/presenter/CategoryOverviewPresenter.kt
package com.robotnec.budget.core.mvp.presenter

import com.robotnec.budget.refactored.di.ApplicationComponent
import com.robotnec.budget.core.domain.Category
import com.robotnec.budget.core.mvp.view.CategoryOverviewView
import com.robotnec.budget.core.persistence.dao.TransactionDao
import com.robotnec.budget.core.service.aggregation.AggregationService
import javax.inject.Inject

/**
 * @author zak zak@swingpulse.com>
 */
class CategoryOverviewPresenter(view: CategoryOverviewView, private val category: Category) : Presenter<CategoryOverviewView>(view) {

    @Inject
    internal lateinit var transactionDao: TransactionDao

    @Inject
    internal lateinit var aggregationService: AggregationService

    override fun injectComponent(applicationComponent: ApplicationComponent) {
        applicationComponent.inject(this)
    }

    override fun onViewResume() {
        val categoryTransactions = transactionDao.getTransactionsForCategory(category.id)
        val aggregation = aggregationService.aggregate(categoryTransactions, AggregationService.Resolution.DAY)
        view.displayCategoryTransactions(aggregation)
        view.displayCategoryTitle(category.name)
    }
}
